deis-cleanup.rb

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env ruby
+
+require "open3"
+require "csv"
+
+KEEP_LAST_VERSIONS = ENV["KEEP_LAST_VERSIONS"].to_i || 2
+DRY_RUN = ENV["DRY_RUN"] || false
+EXCLUDE_REGEX = case ENV["EXCLUDE_REGEX"]
+                when nil then /alpine|deis|blackhole|none|datadog|cleanup|heroku|python/
+                else /#{ENV["EXCLUDE_REGEX"]}/
+                end
+
+def images
+  output, status = Open3.capture2e("docker images")
+  abort output unless status.success?
+
+  headers, *rows = CSV.parse(output, col_sep: " ").map { |e| e.first(2) }
+  headers.map! { |header| header.downcase.to_sym }
+  rows.map { |row| headers.zip(row).to_h }
+end
+
+excluded = -> (image) { image[:repository].match EXCLUDE_REGEX }
+image_type = -> (image) { image[:tag] =~ /git/ ? :git : :versioned }
+repository_name = -> (image) { image[:repository] }
+remove = lambda do |image|
+  command = "docker rmi #{image[:repository]}:#{image[:tag]}"
+  command.prepend "echo " if DRY_RUN
+  output, _status = Open3.capture2e(command)
+  puts output
+end
+
+puts "deis-cleanup started."
+
+filtered = images.reject(&excluded).group_by(&image_type)
+
+%i(git versioned).each do |type|
+  filtered[type] ||= {}
+end
+
+filtered[:git].each(&remove)
+filtered[:versioned].group_by(&repository_name).each do |_, repository|
+  repository.sort_by { |e| e[:tag].split("v").last.to_i }[0..-(KEEP_LAST_VERSIONS + 1)].each(&remove)
+end
```

Comment:
I think your comment is invalid. We don't remove all `versioned` images, we're leaving `KEEP_LAST_VERSIONS`. There is no other types than `git` and `versioned`.